app/redis_client.go

```diff
@@ -0,0 +1,34 @@
+package app
+
+import (
+	"context"
+	"time"
+)
+
+type RedisService struct {
+	repository RedisRepository
+}
+
+func NewRedisService(rs RedisRepository) *RedisService {
```

Comment:
Similarly to how things have been done by now, you can also do this Redis implementation. What I'm saying is, you have a Session model here in the `app` package. Why not put the `AuthSession` model here as well and define an `AuthSessionRepository` (it would have methods such as creating and finding `AuthSession`, just like we've been doing with other models).  You could then implement that interface in the `redis` package.

The only reason I'm saying this is just so we follow the ideas we've been applying until now: separating persistence from app logic. Redis is just another form of persistence, just like PostgreSQL is (just with different properties).